elastic/kibana
# [Discover] Fix filtering selected sidebar fields

## Summary
Field filter didn't filter out selected fields. Instead fields with unknown type icons were displayed. This PR fixes this behavior, filtering fields correctly.

**Test:**
1. Go to Discover
2. Add a number and a string field
3. Filter field list by string
4. Number field is no longer displayed

fixes #91647

### Checklist
